### ISSUE TITLE
Expand Channel Groups for better accessibility 

### ISSUE THREAD

**Issue by whatsforbreakfast** [TeamNewPipe/NewPipe#6372]
### Checklist

- [x] I checked, but didn't find any duplicates (open OR closed) of this issue in the repo. 
- [ x] I have read the contribution guidelines given at https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md.
- [x ] This issue contains only one feature request. I will open one issue for every feature I want to request.

#### Describe the feature you want
<!-- A clear and concise description of what you wish should happen.
It would be really handy if the Channel Groups menu could be expanded by dragging down, increasing the size of the icons. For lots of Groups too it would be better imo if when expanded you can swipe up/down to navigate them instead of side to side.

#### Is your feature request related to a problem? Please describe it
<!-- A clear and concise description of what the problem is. Maybe the developers and the community could brainstorm and come up with a better solution to your problem. If they exist, link to related Issues and/or PRs for developers to keep track easier.

#### Additional context
<!-- Add any other context, like screenshots, about the feature request here.
Here's a picture I mocked up to illustrate what I mean 

![Newpipe channel groups expand](https://user-images.githubusercontent.com/84831769/119608669-ad02fe80-be39-11eb-8382-d029241afc24.png)

#### How will you/everyone benefit from this feature?
<!-- Convince us! How does it change your NewPipe experience and/or your life?
This feature would improve accessibility for those struggling with the small channel groups menu at the moment, right now its very easy for side to side swipes to not register and its very hard to read the group names and tap on the right one each time.

**Comment by whatsforbreakfast** [comment]
![Newpipe channel groups expand](https://user-images.githubusercontent.com/84831769/119608796-e50a4180-be39-11eb-89ab-1cfcb059d01d.png)

 Sorry this is my first time posting a request on github, I think I might have messed up the formatting on the description box. Here's a picture I mocked up to show the change I'd like. It's to improve the accessibility of the Channel Groups menu by adding the ability to drag down and expand it so the icons are larger and easier to read and navigate. I think scrolling down/up for a large collection of groups would work much better too than side to side like it does now. Thanks for your consideration!

**Comment by SameenAhnaf** [comment]
Your suggestion is nice. But the problem with this layout is that users will be forced to view a huge list of channel groups all the time. Some users even have 20 or 30 channel groups and a large portion of screen will be covered all the time.

As an alternative, we may replace "Group alignment" icon by "Secondary controls" icon for channel groups. It's already implemented for "Subscriptions".

https://user-images.githubusercontent.com/79245641/119667820-712b6180-be58-11eb-8e66-d28c231cf3e4.mp4

Single tap on "Secondary controls" icon: Show channel groups in list view. Action holder will be placed on right (just like videos on created playlists) to replace the position of channel groups.
Swipe through channel groups: Show channel groups in grid view as usual.

Unfortunately, gestures are not available on Android TVs and it's quite tricky to make changes into a channel group. So, these actions could be added for channel groups while channel groups are in list view.
Single tap on the channel group icon: Access the channel group long press menu
Single tap on the channel group title bar: Browse the channel group feed as usual.

**Comment by whatsforbreakfast** [comment]
Sorry I think I did a bad job explaining it. I was suggesting keeping the default layout as is, but adding the ability to expand it by either swiping down or with a menu as you suggested which only then would increase the size of the group icons and labels making them more legible, and the area of the screen they now occupy could be scrollable to accommodate large lists of groups. Hope that makes a bit more sense, and here's another illustration to kind of show what I mean
![Newpipe channel groups expand scrollable](https://user-images.githubusercontent.com/84831769/119712128-9f825e80-bea3-11eb-89e9-01b77e9ed459.png)

**Comment by SameenAhnaf** [comment]
>I think scrolling down/up for a large collection of groups would work much better too than side to side like it does now. 

Sorry, I was skimming through the issue. So, I didn't notice everything. Thanks for better visual representation.

However, my suggestions could still be added as I mentioned earlier that Android TVs require a better management.

>Single tap on "Secondary controls" icon: Show channel groups in list view. Action holder will be placed on right (just like videos on created playlists) to replace the position of channel groups.
Swipe through channel groups: Show channel groups in grid view as usual.
>
>And, yes, New (+) icon should be added on the top (of list) or on the left (on grid view) instead of the bottom as they are really hard to access when you are overwhelmed with too many channel groups.
>
>Unfortunately, gestures are not available on Android TVs and it's quite tricky to make changes into a channel group. So, these actions could be added for channel groups while channel groups are in list view.
Single tap on the channel group icon: Access the channel group long press menu
>Single tap on the channel group title bar: Browse the channel group feed as usual.

![IMG_20210527_052024](https://user-images.githubusercontent.com/79245641/119743511-e2483480-beab-11eb-847f-4124a9e5f0c7.jpg)

**Comment by whatsforbreakfast** [comment]
I agree that would be a really cool solution to expanding the channel groups out. Hope we can see something like it come to the app!
